Test.c

```diff
@@ -4,11 +4,156 @@
+      if (index > 12 || strToNum > INT_MAX || strToNum < INT_MIN) isNum = false;
+   }
+   if (isNumCpy != NULL) *isNumCpy = isNum;
+   return str;
+}
+
+char* NumToString (int num, char* input) {
+   char* strNum = malloc (sizeof (char) * 12);
+   if (strNum == NULL) strNum = ERROR_MEM_ALLOC;
+   else if (num == -1 && strcmp (input, "-1") != 0) strNum = "Overflow";
+   else sprintf (strNum, "%d", abs (num));
+   return strNum;
+}
+
+void PrintResult (char* str) {
+   bool* isNum = false;
+   char* input = Filter (str, &isNum);                                     // Filtered input string
+   if (!strcmp (str, "\\n") || !strcmp (str, "\\t") || !strcmp (str, "\\r") || !strcmp (input, EMPTY))
+      printf ("Output =  %s\n", EMPTY"\n"PAL_FALSE);
+   else if (isNum == true) {
+      int num = atoi (input), numRev = ReverseNum (num);
+      char* reverseStr = NumToString (numRev, str);  // Converts the reversed number/ output message to a string
+      printf ("Output = %s%s\n", reverseStr, (num < 0) && strcmp (reverseStr, "Overflow") ? "-" : "");
+      printf ("%s\n", (num >= 0) && (num == numRev) ? PAL_TRUE : PAL_FALSE);
+   } else
+      printf ("%s\n", IsPalindrome (input) ? PAL_TRUE : PAL_FALSE);
+}
+
+void TestReverseNum () {
+   long long int test[8][12] = { { 121 , 121 }, { 1234321 , 1234321 }, { 2147447412 , 2147447412 },
+                               { 123456 , 654321 }, { 2147483646 , -1 },{ -123 , -321 },
+                               { 421124 , 421124 }, { 999999999999 , -1 } };
+   printf ("------------------------------"MAGENTA"Test ReverseNum"RESET
+           "-------------------------------------\n");
+   printf ("   "YELLOW"Input"RESET"      "YELLOW"Expected Output  "RESET" "YELLOW"Actual Output "RESET
+           "\t    "YELLOW"Result\t"RESET"\t"YELLOW"Status"RESET"\n");
+   printf ("----------------------------------------------------------------------------------\n");
+   for (int i = 0; i < 8; i++) {
+      int fOut = ReverseNum (test[i][0]);
+      printf ("%12lld |  %12lld |  %12d |  %18s |    %s\n",
+              test[i][0], test[i][1], fOut,
+              (test[i][0] == fOut && fOut >= 0) ? "Palindrome" : "Not a Palindrome",
+              fOut == test[i][1] ? CYAN"Pass"RESET : MAGENTA"Fail"RESET);
+   }
+   printf ("----------------------------------------------------------------------------------\n\n");
+}
+
+void TestIsPalindrome () {
+   Type test[] = { {"Malayalam",1} , {"Was it a car or a cat I saw",1} , {"Piano",0} ,
+                   {"Sit on a potato pan, Otis!",1} , {"Mr. Owl ate my metal worm.",1} ,
+                   {"Eva, can I see bees in a cave?",1}, {"#2#",1} , {"@#$2",0},{"!;!",1 },{"{;",0 } };
+   int length = sizeof (test) / sizeof (test[0]);
+   printf ("-----------------------------------"MAGENTA"Test ReverseStr"RESET
+           "------------------------------------------\n");
+   printf (""YELLOW "\tInput"RESET"\t\t"YELLOW"\t\tExpected Output"RESET"\t"

```

Comment:
We do not use \t at metamation. Use 3 spaces instead. It ensures consistency across machines.

Reply:
You don't need to set and reset color for every word. Once in a printf statement is enough (unless you're changing colors).